tutorials/2023-05-07-writing-a-fungible-contract-using-python-and-neo3-boa/index.md

```diff
@@ -0,0 +1,356 @@
+
+- `coin.nef`: The contract binary file (bytecode);
+- `coin.manifest.json`: The contract manifest file. This file is required to deploy the contract and is automatically
+generated by the compiler alongside the nef.
+- `coin.nefdbgnfo`: The debug information file. This file is required by the Neo Blockchain Toolkit to debug the contract.
+
+## 5. Running the contract using the Neo Blockchain Toolkit
+
+Click on the `.nef` or `.manifest.json` generated file and press `Ctrl+Shift+D` to open the Run and Debug panel.
+Click on `create a launch.json file` and select `Neo Contract` as the environment.
+
+This will generate a `.vscode/launch.json` file inside the project folder. Open the file and replace the `operation`
+field value with `totalSupply`:
+
+```json
+...
+"invocation": {
+    "operation": "totalSupply",
+    "args": []
+}
+...
+```
+
+Next, press `F5` to start debugging. The Neo Blockchain Toolkit will automatically deploy the contract and invoke the
+`totalSupply` method in a test environment.
+
+Debugging the `totalSupply` method will not be very useful, but it's a good way to test if your environment is working
+properly. After debugging, you should get a message on the debug console showing the GAS consumed and the return value of the method.
+
+## 6. NEP-17 Methods
+
+Since Neo's first supported languages were C#, VB.Net, F#, Java, and Kotlin, the naming convention of the Neo standards
+methods is not snake_case. However, Neo3-boa allows you to use snake_case methods as long as you add the `name`
+parameter to the `public` decorator when necessary. You'll also need to add the `safe=True` parameter on read-only
+methods to indicate that the method won't change the contract storage and can be safely called by other contracts.
+
+The NEP-17 standard defines 5 mandatory methods that a token contract must implement:
+
+### symbol
+
+Returns the token symbol. In this example, we will return the string `COIN`. Since the string `symbol` is the same on
+snake_case and camelCase, we don't need to add the `name` parameter.
+
+```python
+from boa3.builtin.compile_time import public
+
+@public(safe=True)
+def symbol() -> str:
+    return "COIN"
+```
+
+### decimals
+
+Returns the number of decimals used by the token.
+This is used to provide decimal precision when displaying token balances because Neo doesn't support floating types as
+they are often unreliable. In this example, we will return **2**.
+
+```python
+from boa3.builtin.compile_time import public
+
+@public(safe=True)
+def decimals() -> int:
+    return 2
+```
+
+### totalSupply
+
+Returns the total supply of the token. We already implemented this method before, but, since we just defined that there will be
+2 decimals, we need to multiply the supply by 10 ** 2.
+
+```python
+from boa3.builtin.compile_time import public
+
+@public(name='totalSupply', safe=True)
+def total_supply() -> int:
+    return (10 ** 8) * 10 ** decimals()
+```
+
+### balanceOf
```

Comment:
I think we should insert the token on the storage before reading, otherwise it will be too abstract

Reply:
So maybe we should move the _deploy method before balanceOf. Make sure to change the text so the new order makes sense.